web/crux/src/app/team/team.service.ts

```diff
@@ -162,43 +170,124 @@ export class TeamService {
     return CreateEntityResponse.fromJSON(team)
   }
 
-  async updateActiveTeam(request: UpdateActiveTeamRequest): Promise<Empty> {
-    const team = await this.teamRepository.getActiveTeamByUserId(request.accessedBy)
-
+  async updateTeam(request: UpdateTeamRequest): Promise<Empty> {
     await this.prisma.team.update({
       where: {
-        id: team.teamId,
+        id: request.id,
       },
       data: {
         ...request,
+        updatedBy: request.accessedBy,
+        updatedAt: new Date(),
       },
     })
 
     return Empty
   }
 
-  async deleteActiveTeam(request: AccessRequest): Promise<void> {
-    const team = await this.teamRepository.getActiveTeamByUserId(request.accessedBy)
-
-    await this.prisma.team.delete({
+  async deleteTeam(request: IdRequest): Promise<Empty> {
+    const teamWithActiveUsers = await this.prisma.team.findFirst({
+      rejectOnNotFound: false,
       where: {
-        id: team.teamId,
+        id: request.id,
+        users: {
+          some: {
+            active: true,
+          },
+        },
+      },
+      select: {
+        users: {
+          select: {
+            userId: true,
+          },
+        },
       },
     })
+
+    await this.prisma.$transaction(async prisma => {
+      await prisma.team.delete({
```

Comment:
It will fail if you have versions with increased versions. (tested) 

```
  Foreign key constraint failed on the field: `VersionsOnParentVersion_parentVersionId_fkey (index)`
````